wp-content/plugins/core/src/Blocks/Types/Card_Grid/Card_Grid_Model.php

```diff
@@ -0,0 +1,152 @@
+<?php
+declare( strict_types=1 );
+
+namespace Tribe\Project\Blocks\Types\Card_Grid;
+
+use Tribe\Project\Blocks\Types\Base_Model;
+use Tribe\Project\Templates\Components\blocks\card_grid\Card_Grid_Controller;
+use Tribe\Project\Templates\Components\link\Link_Controller;
+use Tribe\Project\Templates\Models\Post_List_Object;
+
+class Card_Grid_Model extends Base_Model {
+	/**
+	 * @return array
+	 */
+	public function get_data(): array {
+		return [
+			Card_Grid_Controller::CLASSES     => $this->get_classes(),
+			Card_Grid_Controller::TITLE       => $this->get( Card_Grid::TITLE, '' ),
+			Card_Grid_Controller::DESCRIPTION => $this->get( Card_Grid::DESCRIPTION, '' ),
+			Card_Grid_Controller::CTA         => $this->get_cta_args(),
+			Card_Grid_Controller::POSTS       => $this->get_posts(),
+			Card_Grid_Controller::LAYOUT      => $this->get( Card_Grid::LAYOUT, Card_Grid::LAYOUT_STACKED ),
+		];
+	}
+
+	/**
+	 * @return array
+	 */
+	private function get_cta_args(): array {
+		$cta = wp_parse_args( $this->get( Card_Grid::CTA, [] ), [
+			'title'  => '',
+			'url'    => '',
+			'target' => '',
+		] );
+
+		return [
+			Link_Controller::CONTENT => $cta[ 'title' ],
+			Link_Controller::URL     => $cta[ 'url' ],
+			Link_Controller::TARGET  => $cta[ 'target' ],
+		];
+	}
+
```

Comment:
From here on down, sure would be nice to have some post list fields models or helper methods so we don't have to copy/paste this every time we want to nab the data for the field, though it does give us the ultimate flexibility to tweak and customize too.

Reply:
Yep! I know that's on the short list of enhancements.